### PR TITLE
ON-37531 # add FormVersion type

### PR DESCRIPTION
**Pull request by divporter**

### CODE REVIEW

File: typescript/forms.d.ts [hunk 1]
```diff
@@ -1225,3 +1221,11 @@ export type FormElementLookupSearchResponse = {
 export type FormElementOptionSetSearchResponse = {
   formElementOptionSets: FormElementOptionSet[]
 } & BaseSearchResult
+
+export type FormVersion = {
+  id: number
+  formId: number
+  form: Form
+  user: UserProfile
+  createdAt: Date
```
Comment (mymattcarroll) [inline typescript/forms.d.ts:1230]:
```suggestion
  createdAt: string
```

File: typescript/forms.d.ts [hunk 2]
```diff
@@ -1225,3 +1221,11 @@ export type FormElementLookupSearchResponse = {
 export type FormElementOptionSetSearchResponse = {
   formElementOptionSets: FormElementOptionSet[]
 } & BaseSearchResult
+
+export type FormVersion = {
+  id: number
+  formId: number
+  form: Form
```
Comment (mymattcarroll) [inline typescript/forms.d.ts:1228]: This should prob omit a few things omitted

File: typescript/forms.d.ts [hunk 3]
```diff
@@ -1225,3 +1221,11 @@ export type FormElementLookupSearchResponse = {
...
+  user: UserProfile
```
Comment (mymattcarroll) [inline typescript/forms.d.ts:1229]: Need to also cater for a key updating a form.

File: typescript/forms.d.ts (hunk 2 above)
Comment (Zaxist) [reply typescript/forms.d.ts:1228]: > "omit a few things omitted"

 Vintage Matt.

Comment (divporter) [reply typescript/forms.d.ts:1228]: why omit anything?

Comment (mymattcarroll) [reply typescript/forms.d.ts:1228]: i am obsessed with removing duplicated data 🤷 I will not die on this hill.

Comment (mymattcarroll) [reply typescript/forms.d.ts:1228]: If we are not going to omit anything, do we need to store `createdAt`? Wouldn't that be `updatedAt` on the form? Assuming we are not going to store the current version in the `FormVersion` table...

**Pull request merged**